code/__HELPERS/icon_smoothing.dm

```diff
@@ -41,18 +41,61 @@
 		if(!AM.anchored)
 			return 0
 
-		for(var/direction in alldirs)
+		for(var/direction in cardinal)
 			AM = find_type_in_direction(A, direction)
-			if(istype(AM))
-				if(AM.anchored)
-					adjacencies |= 1 << direction
-			else
-				if(AM)
-					adjacencies |= 1 << direction
+			if(istype(AM) && AM.anchored)
+				adjacencies |= 1 << direction
+			else if(AM)
+				adjacencies |= 1 << direction
```

Comment:
This happens with most logical operators, it's called Short Circuiting, it's part of Lazy Evaluation.